### PR TITLE
disasm: printed strings can have utf16le byte order mark at front

### PR DESCRIPTION
**Pull request by kazarmy**
Related to https://github.com/radare/radare2/issues/7786#issuecomment-309718837.

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -2909,7 +2910,8 @@ static void ds_print_ptr(RDisasmState *ds, int len, int idx) {
 			}
 		}
 #if 1
-		if (!(IS_PRINTABLE (*msg) || ISWHITECHAR (*msg))) {
+		if (!(IS_PRINTABLE (*msg) || ISWHITECHAR (*msg)
+		      || (len > 1 && !memcmp(msg, "\xff\xfe", 2)))) {
```
Comment (radare) [inline libr/core/disasm.c]: missing space before (

File: libr/core/disasm.c [hunk 2]
```diff
@@ -2705,7 +2705,8 @@ static void ds_print_str(RDisasmState *ds, const char *str, int len) {
 		break;
 	default:
 		str_len = strlen (str);
-		if (str_len == 1 && len > 3 && str[2] && !str[3]) {
+		if ((str_len == 1 && len > 3 && str[2] && !str[3])
+		    || (str_len == 3 && len > 5 && !memcmp(str, "\xff\xfe", 2) && str[4] && !str[5])) {
```
Comment (radare) [inline libr/core/disasm.c]: missing space before (

**Comment by radare** [comment]
tired of waiting, fix those spaces in another pr

**Comment by kazarmy** [comment]
I (try to) follow the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html#Function_Calls) for my own code, which might make me prone to regressing. Tbh, the style guide does make for interesting reading.

> tired of waiting

Please wait another hour. I should be home by then.

> fix those spaces in another pr

See #8668.

**Comment by radare** [comment]
r2 is not c++ and we dont follow the google style guidelines

**Comment by kazarmy** [comment]
That's ok with me ... best to follow as much as possible local conventions
anywhere.

**Comment by Maijin** [comment]
use ./sys/indent.sh

**Comment by Maijin** [comment]
And https://github.com/radare/radare2/blob/master/CONTRIBUTING.md

**Comment by radare** [comment]
aka r2-indent

**Comment by kazarmy** [comment]
> use ./sys/indent.sh
> aka r2-indent

Emacs seems to be doing a good job (i.e. nobody has complained about my indentation yet).

> And https://github.com/radare/radare2/blob/master/CONTRIBUTING.md

I have read that document. I've also noticed that there are a lot of lines > 78 chars in the code so I'm basically formatting my patches according to surrounding code right now. Perhaps I should follow the document more closely.

**Pull request merged**